core/src/main/java/org/aerogear/mobile/core/MobileCore.java

```diff
@@ -236,4 +240,14 @@ public ServiceConfiguration getServiceConfigurationById(final String id) {
         return serviceConfigById.get(id);
     }
 
+    /**
+     * Returns manager that adds new interceptor to the chain of core http interceptors that could
+     * be used to add headers to network requests
+     *
+     * @return RequestHeaderInterceptor
+     */
+    public RequestHeaderInterceptor requestHeaderInterceptor() {
```

Comment:
Should not this be [HttpService's](https://github.com/aerogear/aerogear-android-sdk/blob/master/core/src/main/java/org/aerogear/mobile/core/http/HttpServiceModule.java) responsibility?

Reply:
Apollo is using raw okhttp.

Reply:
Not sure if I follow you here. The sync okhttp instance come from the HttpService => `MobileCore.getInstance().getHttpLayer().getClient()`